### PR TITLE
Separate the tiff encoder classes

### PR DESCRIPTION
**Pull request by IldarKhayrutdinov**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

* Split `TiffWriter` class.
* Move decompressor classes.
 
Only splitting and moving classes, no change algorithms.

### CODE REVIEW

File: src/ImageSharp/Formats/Tiff/Writers/TiffBaseColorWriter.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    internal abstract class TiffBaseColorWriter
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TiffBaseColorWriter" /> class.
+        /// </summary>
+        /// <param name="output">The output stream.</param>
+        /// <param name="memoryAllocator">The memory allocator.</param>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="entriesCollector">The entries collector.</param>
+        public TiffBaseColorWriter(TiffStreamWriter output, MemoryAllocator memoryAllocator, Configuration configuration, TiffEncoderEntriesCollector entriesCollector)
```
Comment (brianpopow) [inline src/ImageSharp/Formats/Tiff/Writers/TiffBaseColorWriter.cs:25]: protected?

File: src/ImageSharp/Formats/Tiff/Writers/TiffStreamWriter.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        public bool IsLittleEndian => BitConverter.IsLittleEndian;
+
+        /// <summary>
+        /// Gets the current position within the stream.
+        /// </summary>
+        public long Position => this.BaseStream.Position;
+
+        public Stream BaseStream { get; }
+
+        protected MemoryAllocator MemoryAllocator { get; }
```
Comment (brianpopow) [inline src/ImageSharp/Formats/Tiff/Writers/TiffStreamWriter.cs:35]: MemoryAllocator and Configuration below seems not to be used. Can they be removed?

**Review by brianpopow — APPROVED** [review]
looks good 👍

File: src/ImageSharp/Formats/Tiff/Writers/TiffBaseColorWriter.cs (hunk 1 above)
Comment (IldarKhayrutdinov) [reply src/ImageSharp/Formats/Tiff/Writers/TiffBaseColorWriter.cs]: sure, done!

File: src/ImageSharp/Formats/Tiff/Writers/TiffStreamWriter.cs (hunk 2 above)
Comment (IldarKhayrutdinov) [reply src/ImageSharp/Formats/Tiff/Writers/TiffStreamWriter.cs]: sure, done!

**Pull request merged**